src/sqlfluff/core/rules/base.py

```diff
@@ -113,28 +113,32 @@ class LintFix:
 
     """
 
-    def __init__(self, edit_type, anchor: BaseSegment, edit=None):
+    def __init__(
+        self,
+        edit_type: str,
+        anchor: BaseSegment,
+        edit: Optional[Iterable[BaseSegment]] = None,
+    ) -> None:
         if edit_type not in (
             "create_before",
             "create_after",
-            "edit",
+            "replace",
             "delete",
         ):  # pragma: no cover
             raise ValueError(f"Unexpected edit_type: {edit_type}")
         self.edit_type = edit_type
         if not anchor:  # pragma: no cover
             raise ValueError("Fixes must provide an anchor.")
         self.anchor = anchor
-        # Coerce to list
-        if isinstance(edit, BaseSegment):
-            edit = [edit]
-        # Copy all the elements of edit to stop contamination.
-        # We're about to start stripping the position markers
-        # of some of the elements and we don't want to end up
-        # stripping the positions of the original elements of
-        # the parsed structure.
-        self.edit = copy.deepcopy(edit)
-        if self.edit:
+        if edit is not None:
+            # Coerce edit iterable to list
+            edit = list(edit)
```

Comment:
iterables all get coerced to lists to keep internal usage unchanged, but rule writers are free to provide whichever format of iterable they choose to the API 👍